crates/cxx-qt-lib/src/core/qcoreapplication.rs

```diff
@@ -87,29 +86,29 @@ impl QCoreApplication {
     /// Prepends path to the beginning of the library path list,
     /// ensuring that it is searched for libraries first.
     /// If path is empty or already in the path list, the path list is not changed.
-    pub fn add_library_path(self: Pin<&mut Self>, path: &QString) {
-        ffi::qcoreapplication_add_library_path(self, path);
+    pub fn add_library_path(path: &QString) {
+        ffi::qcoreapplication_add_library_path(path);
     }
 
     /// The name of this application
-    pub fn application_name(&self) -> QString {
-        ffi::qcoreapplication_application_name(self)
+    pub fn application_name() -> QString {
+        ffi::qcoreapplication_application_name()
     }
 
     /// The version of this application
-    pub fn application_version(&self) -> QString {
-        ffi::qcoreapplication_application_version(self)
+    pub fn application_version() -> QString {
+        ffi::qcoreapplication_application_version()
     }
 
     /// Enters the main event loop and waits until exit() is called,
     /// and then returns the value that was set to exit() (which is 0 if exit() is called via quit()).
-    pub fn exec(self: Pin<&mut Self>) -> i32 {
-        ffi::qcoreapplication_exec(self)
+    pub fn exec(&self) -> i32 {
+        ffi::qcoreapplication_exec()
```

Comment:
Technically `exec` is static too :sweat_smile: although would create a strange API i guess.

And we cannot overload in Rust so we cannot have it be accessible via both :-/

So guess this is the "best" way unless there are other suggestions :thinking: 

https://doc.qt.io/qt-6/qcoreapplication.html#exec

Reply:
Yeah, I wasn't sure of the best approach here. The reason I settled on this is because if it's static, then

```rs
let app = QCoreApplication::new();
QCoreApplication::exec();
```

`app` will be marked by the linter as unused. And I'm not 100% confident that the compiler would know `app` still needs to be alive by the time of `QCoreApplication::exec()`.